### PR TITLE
Updated cycle counts in user manual (merge from cv32e40x)

### PR DESCRIPTION
**Pull request by silabs-oysteink**
Merged updated cycle count chapter and updated for cv32e40s.

### CODE REVIEW

File: docs/user_manual/source/pipeline.rst [hunk 1]
```diff
@@ -68,11 +65,17 @@ and zero stall on the data-side memory interface.
   |                       | 35 (cpuctrl.dataindtiming is set)    | The maximum number of cycles is 35 when the divider is 0    |
   |                       |                                      |                                                             |
   +-----------------------+--------------------------------------+-------------------------------------------------------------+
-  | Jump                  | 3                                    | Jumps are performed in the ID stage. Upon a jump the IF     |
+  | Jump                  | 2 (PC hardening disabled)            | Jumps are performed in the ID stage. Upon a jump the IF     |
```
Comment (Silabs-ArjanB) [inline docs/user_manual/source/pipeline.rst:68]: Branch (not taken) does not seem correct; 1 cycle should be possible. Also update comments

File: docs/user_manual/source/pipeline.rst [hunk 2]
```diff
@@ -68,11 +65,17 @@ and zero stall on the data-side memory interface.
...
   |                       |                                      | stage (including prefetch buffer) is flushed. The new PC    |
-  |                       | 4 (target is a non-word-aligned      | request will appear on the instruction-side memory          |
-  |                       | non-RVC instruction)                 | interface the same cycle the jump instruction is in the ID  |
-  |                       |                                      | stage.                                                      |
+  |                       | 3 (target is a non-word-aligned      | request will appear on the instruction-side memory          |
+  |                       | non-RVC instruction, PC hardening    | interface the same cycle the jump instruction is in the ID  |
+  |                       | disabled)                            | stage.                                                      |
```
Comment (Silabs-ArjanB) [inline docs/user_manual/source/pipeline.rst:72]: the stage -> the IF stage (?)

File: docs/user_manual/source/pipeline.rst (hunk 1 above)
Comment (silabs-oysteink) [reply docs/user_manual/source/pipeline.rst:68]: Fixed, split the untaken branches into enabled/disabled PC hardening.

File: docs/user_manual/source/pipeline.rst (hunk 2 above)
Comment (Silabs-ArjanB) [reply docs/user_manual/source/pipeline.rst:72]: I misread the sentence; it is indeed okay

**Review by Silabs-ArjanB — APPROVED** [review]

**Pull request merged**